Amplify/Categories/Auth/Models/AccessGroup.swift

```diff
@@ -0,0 +1,30 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+
+public struct AccessGroup {
+    public let name: String?
+    public let migrateKeychainItems: Bool
+
+    public init(name: String, migrateKeychainItemsOfUserSession: Bool = false) {
+        self.init(name: name, migrateKeychainItems: migrateKeychainItemsOfUserSession)
```

Comment:
can we enforce checks on the `name`? does apple enforce a format? what happens if customer passes empty string?

Reply:
An incorrect access group name will throw a Keychainstore Error when trying to set or retrieve items (this includes the empty string). Apple does generally enforce a format of "\(teamID).\(bundle_id)". We could enforce it, but if apple changes or adds additional formats this would be breaking.